File: medplum-java/medplum-java-model/src/main/java/com/fhir/JsonWebKey.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableJsonWebKey.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface JsonWebKey {
  Optional<String> qi();

  Optional<uri> implicitRules();

  Optional<String> dq();

  Optional<String> kid();

  Optional<Boolean> active();

  Optional<List<String>> x5c();

  Optional<String> alg();

  Optional<String> n();

  Optional<Meta> meta();

  Optional<String> x5t();

  Optional<code> language();

  Optional<String> dp();

  Optional<List<String>> key_ops();

  Optional<String> p();

  Optional<String> q();

  Optional<id> id();

  Optional<String> use();

  Optional<String> resourceType();

  Optional<String> kty();

  Optional<String> e();

  Optional<String> d();

  static ImmutableJsonWebKey.Builder builder() {
    return ImmutableJsonWebKey.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/JsonWebKey.ts
import { code } from "./code";
import { id } from "./id";
import { Meta } from "./Meta";
import { uri } from "./uri";

export interface JsonWebKey {
    resourceType: string | null | undefined;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    active: boolean | null | undefined;
    alg: string | null | undefined;
    kty: string | null | undefined;
    use: string | null | undefined;
    key_ops: string[] | null | undefined;
    x5c: string[] | null | undefined;
    n: string | null | undefined;
    e: string | null | undefined;
    kid: string | null | undefined;
    x5t: string | null | undefined;
    d: string | null | undefined;
    p: string | null | undefined;
    q: string | null | undefined;
    dp: string | null | undefined;
    dq: string | null | undefined;
    qi: string | null | undefined;
}
